sdk/communication/Azure.Communication.Common/src/CommunicationIdentifier.cs

```diff
@@ -61,7 +61,13 @@ public static CommunicationIdentifier FromRawId(string rawId)
 
             var segments = rawId.Split(':');
             if (segments.Length < 3)
+            {
+                if (segments.Length == 2 && segments[0] == "28")
+                {
+                    return new MicrosoftBotIdentifier(segments[1], true, CommunicationCloudEnvironment.Public);
+                }
                 return new UnknownIdentifier(rawId);
+            }
```

Comment:
The result of this operations could not be consistent if there is another `:` in the identifier. The current approach is consistent across all platforms and will use the remaining part of the string (no matter if it has `:` in it or not). The approach you're proposing will only return the third element of the initial list (no matter if there are more elements in it). this might fail in the future with other identifiers.

Reply:
we have a fixed MRI format and so far there was no change, even if there will be another identifier it will fall under the unknown identifier and there are checks for the known one by prefixes, and we're pretty sure about their formats(as the third element is id) and for them should not be any changes.

Reply:
Ok

Reply:
I'm not quite convinced about this change.
> even if there will be another identifier it will fall under the unknown identifier

@AikoBB yes but we might be sending an incomplete payload to the UnknownIdentifier's constructor... e.g., what if a future identifier will have more than 3 segments?

Reply:
the unknown identififer is initiated by the rawid as follow:
`new UnknownIdentifier(rawId)`

I guess it was your concern and we not providing here the suffix

Reply:
yeah, then it's fine I guess. I thought it was also being initialized using the suffix, sorry for missing that.

Reply:
playing a devils advocate here - what about the case when a new `MicrosoftTeamsUserIdentifier` format will be introduced with more than 3 segments?

Reply:
and I honestly hope this will not happen :)

Reply:
good point, even for known identifiers if there will be introduced another segment we have to identify which part will the id part by releasing a new version, so makes sense to extend the check in this [line](https://github.com/Azure/azure-sdk-for-net/blob/b1a08d0b333e0a30edd12a46d4a22e3dd8c944d7/sdk/communication/Azure.Communication.Common/src/CommunicationIdentifier.cs#L79) maybe, wdyt?

Reply:
> we have to identify which part will the id part by releasing a new version
exactly

>extend the check in this [line](https://github.com/Azure/azure-sdk-for-net/blob/b1a08d0b333e0a30edd12a46d4a22e3dd8c944d7/sdk/communication/Azure.Communication.Common/src/CommunicationIdentifier.cs#L79) maybe

yeah and add a couple more tests for the scenarios we've just described

Reply:
OK